bindings/profilers/heap.cc

```diff
@@ -47,15 +52,61 @@ enum CallbackMode {
 };
 
 struct HeapProfilerState {
-  uint32_t       heap_extension_size;
-  uint32_t       max_heap_extension_count;
-  uint32_t       current_heap_extension_count;
-  uv_async_t     async;
+  explicit HeapProfilerState(v8::Isolate* isolate) : isolate(isolate) {}
+
+  ~HeapProfilerState() {
+    UninstallNearHeapLimitCallback();
+    if (async) {
+      uv_close(reinterpret_cast<uv_handle_t*>(async),
+        [](uv_handle_t* handle){
+          delete reinterpret_cast<uv_handle_t*>(handle);
```

Comment:
It's already a `uv_handle_t`. I assume you're wanting to reinterpret to a `uv_async_t` before deleting?